CLAUDE.md

```diff
@@ -0,0 +1,135 @@
+cargo run --bin evm-testnet
+
+# Create local test network with 25 nodes
+cargo run --bin antctl -- local run --build --clean --rewards-address <YOUR_ETHEREUM_ADDRESS>
+
+# Check node status
+cargo run --bin antctl -- status
+
+# Upload files (requires SECRET_KEY environment variable)
+SECRET_KEY=<YOUR_EVM_SECRET_KEY> cargo run --bin ant -- --local file upload <path>
+
+# Download files
+cargo run --bin ant -- --local file download <addr> <dest_path>
+
+# Tear down network
+cargo run --bin antctl -- local kill
+```
+
+### Binary Builds
+```bash
+# Build release binaries using justfile
+just build-release-artifacts <architecture>
+
+# Supported architectures: x86_64-pc-windows-msvc, x86_64-apple-darwin, 
+# aarch64-apple-darwin, x86_64-unknown-linux-musl, arm-unknown-linux-musleabi,
+# armv7-unknown-linux-musleabihf, aarch64-unknown-linux-musl
+```
+
+## Workspace Architecture
+
+### Core Node Components
+- **ant-node**: The backbone node implementation that stores data and validates transactions
+- **ant-networking**: P2P networking layer built on libp2p with Kademlia DHT
+- **ant-protocol**: Network protocol definitions and message types
+- **ant-node-manager**: Tools for managing local networks and node processes
+
+### Client Components  
+- **autonomi**: Main client API library with high-level data operations
+- **ant-cli**: Command-line interface for end users
+- **nodejs**: Node.js bindings for the client API
+- **autonomi/python**: Python bindings for the client API
+
+### Supporting Components
+- **ant-bootstrap**: Network bootstrap peer discovery and caching
+- **ant-logging**: Centralized logging infrastructure using tracing
+- **ant-metrics**: OpenMetrics exporter for monitoring and observability
+- **evmlib**: EVM blockchain integration for payments
+- **ant-evm**: Payment processing and blockchain interaction
+
+### Development Tools
+- **node-launchpad**: TUI application for managing nodes
+- **nat-detection**: NAT traversal and network connectivity detection
+- **test-utils**: Shared testing utilities across the workspace
+
+## Key Development Patterns
+
+### Testing Strategy
+- Unit tests are run per package due to different requirements and network dependencies
+- E2E tests require a running local network (set up via antctl local run)
+- Integration tests use the ant-local-testnet-action for GitHub Actions
+- Churn tests simulate network instability and node restarts
+
+### Payment System Integration
+All data operations require EVM payments. When developing:
+- Use the local EVM testnet for development (`cargo run --bin evm-testnet`)
+- Set SECRET_KEY environment variable for client operations
+- Payments are handled through the evmlib crate
+
+### Network Architecture
+- Nodes form a Kademlia DHT for routing and data location
+- Data is stored as encrypted chunks across multiple nodes
+- QUIC is the primary transport protocol
+- Network discovery uses bootstrap peers from ant-bootstrap
```

Comment:
that's `initial` network discovery
there is also `peroidical network discovery` (on node side only), which is achieved via querying certain network distance space with a round-robin style.